src/playground-helper.ts

```diff
@@ -0,0 +1,31 @@
+import {compressToEncodedURIComponent} from 'lz-string'
+
+function unindent(value: string) {
+  // remove white spaces first, to save a few bytes.
+  // testing-playground will reformat on load any ways.
+  return value.replace(/[ \t]*[\n][ \t]*/g, '\n')
+}
+
+function encode(value: string) {
+  return compressToEncodedURIComponent(unindent(value))
+}
+
+function getPlaygroundUrl(element: Element | null, logErrors = true) {
+  if (!element || !('innerHTML' in element)) {
```

Comment:
```suggestion
  if (element === null) {
```
Should be sufficient looking at the type definitions.

Reply:
The original if-statement in `logTestingPlaygroundURL` in `screen.js` look like this:

```javascript
if (!element || !('innerHTML' in element))
```

So changing it will cause a behaviour change for that function. Are we okay with that?

Reply:
Considering we now have the type-safety I'm ok with it. Unless the types are inaccurate.

Reply:
Hmm.. 🤔
In `screen.d.ts` the type for the element parameter is `Element | HTMLDocument`. However `HTMLDocument` does not have an `innerHTML` property.

It's not really clear to me why `logTestingPlaygroundURL` accepts a `HTMLDocument` if it requires `innerHTML` at the same time.

Should I keep `Element | null` or is `Element | HTMLDocument | null` the correct type?
Perhaps the type definition in `screen.d.ts` needs to be updated.

Reply:
Another question: do we even want the `| null`? From the type definition it sounds like `null` is a valid value, when in fact it's an erroneous value to give as parameter.
If we do not have `| null` though, we will have to remove the first check or shut up ESLint with:
```
// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
if (!element) {
```

Reply:
### Update:

- I have decided to remove the `| null`, so we're not "advertising" that `null` can be used as a valid value.
- I don't think  `HTMLDocument` is a valid type to use here, so to me it looks like that was an error in the old type definition.
- I have removed a test that tried using `{}` as an argument and expecting `The element you're providing isn't a valid DOM element`. That error will no longer be logged if you're passing `{}`, because I removed the `!('innerHTML' in element)` check, and rely on static type checking instead.

Reply:
> That error will no longer be logged if you're passing {}, because I removed the !('innerHTML' in element) check, and rely on static type checking instead.

We generally don't relly on static type checking for users of our library. We may change that in the future but for now additional runtime errors should not be removed.

Reply:
> That error will no longer be logged if you're passing {}, because I removed the !('innerHTML' in element) check, and rely on static type checking instead.

We generally don't relly on static type checking for users of our library. We may change that in the future but for now additional runtime errors should not be removed.

Reply:
Thanks for the explanation, I think I get it now! I have added the runtime check again in `logTestingPlaygroundURL`.